gateway-shell/src/main/java/org/apache/knox/gateway/shell/KnoxSh.java

```diff
@@ -155,6 +170,81 @@ public boolean validate() {
     public abstract String getUsage();
   }
 
+  private class KnoxBuildTrustStore extends Command {
+
+    private static final String USAGE = "buildTrustStore --gateway server-url";
+    private static final String DESC = "Downloads the gateway server's public certificate and builds a trust store.";
+    private static final String CLIENT_TRUST_STORE_FILE_NAME = "gateway-client-trust.jks";
+
+    @Override
+    public void execute() throws Exception {
+      final X509Certificate gatewayServerPublicCert = fetchPublicCertFromGatewayServer();
+      if (gatewayServerPublicCert != null) {
+        final File trustStoreFile = new File(System.getProperty("user.home"), CLIENT_TRUST_STORE_FILE_NAME);
+        X509CertificateUtil.writeCertificateToJks(gatewayServerPublicCert, trustStoreFile);
+        out.println("Gateway server's certificate is exported into " + trustStoreFile.getAbsolutePath());
+      } else {
+        out.println("Could not obtain server certificate chain");
```

Comment:
Will this happen in a non exception case? Assuming an exception is thrown from fetchPublicCertFromGatewayServer it would propagate up and this message wouldn't be printed.

Reply:
Fixing...